Source/Applications/openXDA/openXDA/Controllers/Workbench/DataFileController.cs

```diff
@@ -325,6 +329,40 @@ IEnumerable<DataRow> QueryFileGroupIDs(IEnumerable<int> ids)
             }
         }
 
+        [Route("Download/{fileID:int}"), HttpGet]
+        public HttpResponseMessage DownloadFile(int fileID)
+        {
+            using (AdoDataConnection connection = NodeHost.CreateDbConnection())
+            {
+                TableOperations<DataFile> fileGroupTable = new TableOperations<DataFile>(connection);
+                DataFile file = fileGroupTable.QueryRecordWhere("ID = {0}", fileID);
+                
+                if (file is null)
+                    throw new InvalidDataException("No file found");
```

Comment:
This seems like a good opportunity to return a 404 Not Found rather than 500 Internal Server Error.

```suggestion
                if (file is null)
                    return await NotFound().ExecuteAsync(default);
```

Reply:
Is a 404 really appropriate in this case? a 404 implies it hasn't found the endpoint vs this is really more of an issue of not having a valid ID/ that ID not existing.

Reply:
I think it's appropriate. A 404 is really "resource not found". The resource in this case would be the file with the given ID. If the `DataFile` record for that ID does not exist, then the resource is not found.